src/test/java/com/library/controller/BookControllerTest.java

```diff
@@ -0,0 +1,208 @@
+package com.library.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.library.common.TestStorage;
+import com.library.model.dto.BookDto;
+import com.library.model.entity.Book;
+import com.library.model.mapper.BookMapper;
+import com.library.repository.BookRepository;
+import com.library.utils.IsSameLikeBook;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.SneakyThrows;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookMapper bookMapper;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    private TestStorage testStorage;
+
+    private static final String MISSING_TYPE = "Not entered data: [authors]";
+
+    private static final String DELETED_SUCCESS = "Book is deleted successfully!";
+
+    @BeforeEach
```

Comment:
Why is testStorage initialized before each test?

Reply:
Because "every test must be independent of others")